aten/src/ATen/native/cuda/ReplicationPadding.cu

```diff
@@ -123,7 +131,13 @@ __global__ void replication_pad_backward_kernel(
   int inputPointY = imin(imax(padT, outputPointY), gradInput.size(2) + padT - 1) - oStartY + iStartY;
 
   scalar_t valueToCopy = gradOutput[batch][plane][outputPointY][outputPointX];
-  gpuAtomicAdd(&gradInput[batch][plane][inputPointY][inputPointX], valueToCopy);
+  fastAtomicAdd(
+    gradInput.data(),
+    idx_2d(batch * channel + plane, inputPointY, inputPointX, height, width),
```

Comment:
packedTensorAccessor supports non-contiguous tensors, yet here you are replacing index computation with index computation that's only valid for contiguous tensors. I don't immediately see if gradInput is guaranteed to be contiguous, probably not.

Reply:
We can change those `channel`, `height`, `width` to be corresponding tensor strides and slightly modify the `idx_2d` function. How does that sound?

Otherwise, we may have to add `gradInput.contiguous()` which may affect performance.